lib/internal/modules/esm/loader.js

```diff
@@ -21,6 +23,14 @@ const {
 } = require('internal/modules/esm/utils');
 let defaultResolve, defaultLoad, defaultLoadSync, importMetaInitializer;
 
+const wellKnownCommunityLoaders = {
+  '__proto__': null,
+  'typescript': [
+    'ts-node',
+    'sucrase',
+  ],
```

Comment:
Before shipping this, we need some kind of policy for how to define what goes into this list. That could maybe be part of the documentation.

Reply:
On further thought, I think the error message should just link to something in https://nodejs.org/en/docs/guides. Then we can have a longer list with detailed instructions. It also protects us against one of these recommendations becoming outdated, like if one of the packages in this list goes unmaintained, we don’t want old versions of Node continuing to recommend it forever.

Reply:
I think that could be good for a v0.1 but isn't sufficient to be our target.

In my presentation on Monday, I mentioned potentially retrieving this registry from a remote endpoint, which would address the “outdated” concern. Antoine pointed out that node currently does not itself make any http requests, so this would be quite unusual. So I was thinking perhaps we could ask package managers for a very slight augmentation to their existing search functionality to support searching for loaders (ex `npm search typescript --searchopts …`), which avoids reinventing the wheel and leverages a familiar experience; also, these seems an appropriate place for that functionality. I messaged @arcanis on Slack to check the temperature of the room, but haven't heard back yet.

Reply:
I think having Node make an HTTP call will be a very hard sell, especially for this use case. It was proposed to have a command like `node --check-version` to see if Node was running the latest version and that got shot down because of security concerns.

You could perhaps include in the prompt some text like “Run `npx setup-typescript` and we can publish a `setup-typescript` command to npm, that starts a wizard and prompts people through it, etc. I would still include a link to the docs, though, so that the only choice isn’t to just run unknown code.

Reply:
I heard back from Maël that the search we're looking for is actually already possible. I checked earlier tonight, and it seems this could do:

```sh
$ npm search --json module-translator typescript
```

We would need a specific tag for whatever we call this packages. `module-translator` appears to be unused so far: https://www.npmjs.com/search?q=keywords:module-translator